core-rust/state-manager/src/staging/cache.rs

```diff
@@ -0,0 +1,146 @@
+ * undertaking, as to the functionality security use, value or other characteristics of
+ * any distributed ledger nor in respect the functioning or value of any tokens which may
+ * be created stored or transferred using the Work. The Licensor does not warrant that the
+ * Work or any use of the Work complies with any law or regulation in any territory where
+ * it may be implemented or used or that it will be appropriate for any specific purpose.
+ *
+ * Neither the licensor nor any current or former employees, officers, directors, partners,
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+use super::stage_tree::{StagedSubstateStoreKey, StagedSubstateStoreNodeKey};
+use crate::staging::stage_tree::{StagedSubstateStore, StagedSubstateStoreManager};
+use crate::AccumulatorHash;
+use radix_engine::ledger::ReadableSubstateStore;
+use radix_engine::transaction::TransactionReceipt;
+use sbor::rust::collections::HashMap;
+use slotmap::SecondaryMap;
+
+pub struct ExecutionCache<S: ReadableSubstateStore> {
```

Comment:
I'm not sold on this name - perhaps `StagedResultsCache` ?

Reply:
+1 to wrapping of the staged store.

Reply:
hmm, when it was purely a "hashmap-slotmap key mapper", the `ExecutionCache` indeed seemed a bit random.
But it does have an `execute()` method right now, and it caches these executions, so the original naming suddenly clicked (to me).

I agree that its _final_ name still might differ, but it depends on how we evolve it (i.e. based on your https://github.com/radixdlt/babylon-node/pull/280#discussion_r1092102002).

Reply:
Fair enough - that makes sense.

I just just thinking about ways to distinguish it from the `pending_transaction_results_cache` - which could also be described as an execution cache. But I don't feel too strongly 👍